### DISCUSSION TITLE
v4 feedback

### DISCUSSION THREAD

**Discussion by SanderMertens** [SanderMertens/flecs#1181] [Ideas]
Post your v4 ideas/feedback here!

**Comment by xgalaxy** [comment]
Late to the party here. I haven't looked at flecs in a long time, way before script was added. A little bit of feedback I have about it is that it looks like you have to make sure to keep your C/C++ components consistent with their FlecsScript equivalent and I'm wondering if there is a way to codegen either the C/C++ side or the FlecsScript side based on the definitions in the other?

It seems to me that script was added to make it much faster and easier to iterate on the component designs but isn't that flexibility lost if you have to manually constantly update the C/C++ side as well? 

Maybe I'm missing something or not grokking fully how it works.

**Reply by SanderMertens** [reply]
> It seems to me that script was added to make it much faster and easier to iterate on the component designs

The primary use case for Flecs script is to quickly iterate on scenes/configuration, not necessarily component types. You _can_ create component types in Flecs script, but that's more a side effect of the reflection system using entities/components to store type data :)

I do agree that if you want to use flecs script for components, it'd be nice to have code generation. That's definitely something you can build today and would be useful to have.

**Reply by xgalaxy** [reply]
Thanks for the reply. I think I understand now. I mistook what I was seeing. Looking closer at the v4 diff for the tower defense game (even though it is a complicated example) helped clear up some confusion. Seeing the removal of the big block of defining prefabs in C++ and doing them in the script is certainly nice.
